src/Logger/LoggingTrait.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Copyright 2024 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Auth\Logger;
+
+trait LoggingTrait
+{
+    /**
+     * @param LogEvent $event
+     *
+     * @return void
+     */
+    private function logRequest(LogEvent $event): void
+    {
+        $timestamp = $event->timestamp;
+
+        $debugEvent = [
+            'timestamp' => $timestamp,
+            'severity' => 'DEBUG', //Perhaps have something like Logger::debug
+        ];
+
+        $jsonPayload = [
+            'request.method' => $event->method,
+            'request.url' => $event->url,
+            'request.headers' => $event->headers,
+            'request.payload' => $event->payload,
+            'request.rpcName' => $event->rpcName,
+            'jwt' => $this->getJwtToken($event->headers),
```

Comment:
the way this is defined here, it will look like this in the payload:

```php
[
    'jwt' => [
        'jwt' => [
            'header' => [ /** ... */ ],
            'payload' => [ /** ... */ ],
        ]
    ]
]
```

is that intentional?